packages/circuit-ui/components/PhoneNumberInput/PhoneNumberInput.tsx

```diff
@@ -273,21 +286,19 @@ export const PhoneNumberInput = forwardRef<
         return;
       }
 
-      const pastedCountryCode = countryCode.options
-        .map((option) => option.code)
-        // Match longer, more specific country codes first
-        .sort((a, b) => b.length - a.length)
-        .find((code) => pastedText.startsWith(code));
+      const pastedOption = countryCode.options.find(({ code }) =>
+        pastedText.startsWith(code),
+      );
```

Comment:
Here, the lookup is done by country code, so the sorting by length remains necessary.

```suggestion
      const pastedOption = countryCode.options
        // Match longer, more specific country codes first
      	.sort((a, b) => b.code.length - a.code.length)
        .find(({ code }) => pastedText.startsWith(code));
```